### PR TITLE
Update github issue downloader

### PR DESCRIPTION
**Pull request by maryamariyan**
- Also downloads issues with null description
- Ignores tabbing to fit each issue in row
- Saves the ID and will strip that as post processing\
- Sleeps for one hour once RateLimitExceededException occurs
- Helps with training to save old SocketsHttpHandler label as net.http

Note,
I am saving the ID for each row, but I just prefer this way, because the removal of first column is a very easy post processing step, easier to keep and to remove later than not have to start with.

### CODE REVIEW

File: src/Microsoft.DotNet.Github.IssueLabeler/GithubIssueDownloader.cs [hunk 1]
```diff
@@ -35,53 +38,81 @@ public GithubIssueDownloader(string authToken, string repoName, string owner, in
             {
                 Credentials = new Credentials(authToken)
             };
-
+
```
Comment (Anipik) [inline src/Microsoft.DotNet.Github.IssueLabeler/GithubIssueDownloader.cs]: nit: extra space

File: src/Microsoft.DotNet.Github.IssueLabeler/GithubIssueDownloader.cs [hunk 2]
```diff
@@ -35,53 +38,81 @@ public GithubIssueDownloader(string authToken, string repoName, string owner, in
...
+                        filePaths = String.Join(";", prFiles.Select(x => x.FileName));
                     }
+                }
+                catch (NotFoundException)
+                {
+                    continue;
+                }
+                catch (RateLimitExceededException ex)
+                {
+                    TimeSpan timeToWait = ex.Reset.AddMinutes(1) - DateTimeOffset.UtcNow;
```
Comment (Anipik) [inline src/Microsoft.DotNet.Github.IssueLabeler/GithubIssueDownloader.cs]: can we just add 5 minutes wait or something ?

File: src/Microsoft.DotNet.Github.IssueLabeler/GithubIssueDownloader.cs [hunk 3]
```diff
@@ -35,53 +38,81 @@ public GithubIssueDownloader(string authToken, string repoName, string owner, in
...
+            if (sb.Length != 0)
+            {
+                File.AppendAllText(_outputFile, sb.ToString());
+            }
+        }
 
-            File.AppendAllText(_outputFile, sb.ToString());
+        private string IgnoreForTraining(string column)
+        {
+            if (_skipColumns) return string.Empty;
```
Comment (Anipik) [inline src/Microsoft.DotNet.Github.IssueLabeler/GithubIssueDownloader.cs]: can we use tertiary operator here ?

File: src/Microsoft.DotNet.Github.IssueLabeler/GithubIssueDownloader.cs [hunk 4]
```diff
@@ -35,53 +38,81 @@ public GithubIssueDownloader(string authToken, string repoName, string owner, in
...
                     }
+                }
+                catch (NotFoundException)
+                {
+                    continue;
+                }
+                catch (RateLimitExceededException ex)
+                {
+                    TimeSpan timeToWait = ex.Reset.AddMinutes(1) - DateTimeOffset.UtcNow;
+                    await Task.Delay((int)timeToWait.TotalMilliseconds).ConfigureAwait(false);
```
Comment (ChadNedzlek) [inline src/Microsoft.DotNet.Github.IssueLabeler/GithubIssueDownloader.cs]: Task.Delay takes a TimeSpan... why force it back into the millisecond one?

**Pull request merged**